go/extractor/util/registryproxy.go

```diff
@@ -71,47 +96,43 @@ func checkEnvVars() {
 		} else {
 			// We only care about private registry configurations that are relevant to Go and
 			// filter others out at this point.
-			proxy_configs = make([]RegistryConfig, 0)
 			for _, cfg := range val {
 				if cfg.Type == GOPROXY_SERVER {
 					proxy_configs = append(proxy_configs, cfg)
 					slog.Info("Found GOPROXY server", slog.String("url", cfg.URL))
 				}
 			}
+
+			if len(proxy_configs) > 0 {
+				goproxy_val := "https://proxy.golang.org,direct"
+
+				for _, cfg := range proxy_configs {
+					goproxy_val = cfg.URL + "," + goproxy_val
+				}
+
+				result = append(result, fmt.Sprintf("GOPROXY=%s", goproxy_val), "GOPRIVATE=", "GONOPROXY=")
+			}
 		}
 	}
 }
 
 // Applies private package proxy related environment variables to `cmd`.
 func ApplyProxyEnvVars(cmd *exec.Cmd) {
-	slog.Info(
+	slog.Debug(
 		"Applying private registry proxy environment variables",
 		slog.String("cmd_args", strings.Join(cmd.Args, " ")),
 	)
 
-	checkEnvVars()
-
-	// Preserve environment variables
-	cmd.Env = os.Environ()
-
-	if proxy_address != "" {
-		cmd.Env = append(cmd.Env, fmt.Sprintf("HTTP_PROXY=%s", proxy_address))
-		cmd.Env = append(cmd.Env, fmt.Sprintf("HTTPS_PROXY=%s", proxy_address))
+	// If we haven't done so yet, check whether the proxy environment variables are set
+	// and extract information from them.
+	if !proxy_vars_checked {
+		proxy_vars = getEnvVars()
+		proxy_vars_checked = true
 	}
-	if proxy_cert_file != "" {
-		slog.Info("Setting SSL_CERT_FILE", slog.String("proxy_cert_file", proxy_cert_file))
-		cmd.Env = append(cmd.Env, fmt.Sprintf("SSL_CERT_FILE=%s", proxy_cert_file))
-	}
-
-	if len(proxy_configs) > 0 {
-		goproxy_val := "https://proxy.golang.org,direct"
-
-		for _, cfg := range proxy_configs {
-			goproxy_val = cfg.URL + "," + goproxy_val
-		}
 
-		cmd.Env = append(cmd.Env, fmt.Sprintf("GOPROXY=%s", goproxy_val))
-		cmd.Env = append(cmd.Env, "GOPRIVATE=")
-		cmd.Env = append(cmd.Env, "GONOPROXY=")
+	// If the proxy is configured, `proxy_vars` will be not `nil`. We append those
+	// variables Preserve environment variables
```

Comment:
There is a stray "Preserve environment variables" at the end of this comment.

Relatedly, I see that if `proxy_vars` is `nil` then `cmd.Env` is not set, and this means that the parent's environment variables are used. Please add a comment that makes this clear, as it was not immediately obvious to me.

Reply:
I think this must have happened as a result of an automatically resolved merge conflict or something. I'll fix it, thanks for spotting it!